cdc/redo/manager.go

```diff
@@ -98,302 +160,171 @@ func (s *statefulRts) setFlushed(flushed model.Ts) {
 		}
 	}
+	return true
 }
 
-// ManagerImpl manages redo log writer, buffers un-persistent redo logs, calculates
-// redo log resolved ts. It implements LogManager interface.
-type ManagerImpl struct {
-	changeFeedID model.ChangeFeedID
-	enabled      bool
+// logManager manages redo log writer, buffers un-persistent redo logs, calculates
+// redo log resolved ts. It implements DDLManager and DMLManager interface.
+type logManager struct {
+	enabled bool
+	cfg     *writer.LogWriterConfig
+	writer  writer.RedoLogWriter
 
-	opts *ManagerOptions
+	rwlock sync.RWMutex
+	// TODO: remove logBuffer and use writer directly after file logWriter is deprecated.
+	logBuffer *chann.Chann[cacheEvents]
+	closed    int32
 
 	// rtsMap stores flushed and unflushed resolved timestamps for all tables.
 	// it's just like map[span]*statefulRts.
 	// For a given statefulRts, unflushed is updated in routine bgUpdateLog,
 	// and flushed is updated in flushLog.
 	rtsMap spanz.SyncMap
 
-	// A fast way to get min flushed timestamp in rtsMap. It can be updated
-	// in redo-flush goroutine and AddTable/RemoveTable.
-	minResolvedTs uint64
-
-	// Updated by `UpdateResolvedTs`. It's required by some internal places,
-	// such as GC.
-	checkpointTs uint64
-
-	metaCheckpointTs statefulRts
-	metaResolvedTs   statefulRts
-
-	rwlock    sync.RWMutex
-	writer    writer.RedoLogWriter
-	logBuffer *chann.DrainableChann[cacheEvents]
-	closed    int32
-
-	flushing      int64
-	lastFlushTime time.Time
+	flushing         int64
+	lastFlushTime    time.Time
+	releaseMemoryCbs []func()
 
 	metricWriteLogDuration    prometheus.Observer
 	metricFlushLogDuration    prometheus.Observer
 	metricTotalRowsCount      prometheus.Counter
 	metricRedoWorkerBusyRatio prometheus.Counter
 }
 
-// NewManager creates a new Manager
-func NewManager(ctx context.Context, cfg *config.ConsistentConfig, opts *ManagerOptions) (*ManagerImpl, error) {
+func newLogManager(
+	ctx context.Context, cfg *config.ConsistentConfig, logType string,
+) (*logManager, error) {
 	// return a disabled Manager if no consistent config or normal consistent level
 	if cfg == nil || !redo.IsConsistentEnabled(cfg.Level) {
-		return &ManagerImpl{enabled: false}, nil
+		return &logManager{enabled: false}, nil
 	}
 
-	writer, err := writer.NewRedoLogWriter(ctx, cfg, opts.FileTypeConfig)
+	uri, err := storage.ParseRawURL(cfg.Storage)
 	if err != nil {
 		return nil, err
 	}
-
-	m := &ManagerImpl{
-		changeFeedID:  contextutil.ChangefeedIDFromCtx(ctx),
-		enabled:       true,
-		opts:          opts,
-		rtsMap:        spanz.SyncMap{},
-		minResolvedTs: math.MaxInt64,
-		writer:        writer,
-	}
-
-	if m.opts.EmitMeta {
-		checkpointTs, resolvedTs := m.writer.GetMeta()
-		m.metaCheckpointTs.flushed = checkpointTs
-		m.metaCheckpointTs.unflushed = checkpointTs
-		m.metaResolvedTs.flushed = resolvedTs
-		m.metaResolvedTs.unflushed = resolvedTs
+	changefeedID := contextutil.ChangefeedIDFromCtx(ctx)
+	m := &logManager{
+		enabled: true,
```

Comment:
If it's always true, can we remove it?

Reply:
ref [L209](https://github.com/pingcap/tiflow/pull/8107/files#diff-ab3a1fba521031cda0725a505d31255067c9d321df8059ce6150a7a5b2022968R209)